transports/noise/src/io/handshake.rs

```diff
@@ -91,10 +106,42 @@ impl<T> State<T> {
             return Err(Error::BadSignature);
         }
 
+        // Check WebTransport certhashes that responder reported back to us
+        if is_initiator {
+            // We check only if we care (i.e. Config::with_webtransport_certhashes was used)
+            if let Some(valid_certhashes) = self.responder_webtransport_certhashes {
+                let ext = self
+                    .remote_extensions
+                    .ok_or(Error::BadWebTransportCerthashes)?;
+
+                // The known WebTransport certhashes must be a strict subset
+                // of the reported ones
+                let is_valid = valid_certhashes
+                    .iter()
+                    .all(|hash| ext.webtransport_certhashes.contains(hash));
+
+                if !is_valid {
+                    return Err(Error::BadWebTransportCerthashes);
```

Comment:
```suggestion
                    return Err(Error::UnknownWebTransportCerthashes);
```

Reply:
Might also be worth reporting, which one we rejected / did not recognise. That would make debugging a lot easier I imagine.

Reply:
Should I use b58 to display them?

Reply:
Yes that sounds reasonable. `Multihash` should have a function for that I believe?

Also, if you switch to `HashSet`s, you can make use of `HashSet::difference` to conveniently figure out which ones are missing.

Reply:
> Yes that sounds reasonable. Multihash should have a function for that I believe?

It doesn't have, I will add bs58 in libp2p-noise for now